packages/ingest/test/test-S3ProviderClient.js

```diff
@@ -110,6 +108,26 @@ test.serial('S3ProviderClient.sync syncs a file to a target S3 location', async
   );
 });
 
+test.serial('S3ProviderClient.sync syncs a 0 byte file', async (t) => {
```

Comment:
This test doesn't really prove anything since it passed against Localstack even before the fixes were made. However, in case Localstack does ever update their multipart upload handling to match real S3 behavior, this will be a good test to have.

Reply:
Would it make sense to have this test check that `s3CopyObject ` was used in place of `multipartCopyObject` when a 0 byte files was uploaded? Tests really shouldn't know about the function's implementation like that but maybe it would be worthwhile to ensure that we're using the new correct code path if this is always going to work locally?

Thinking out loud: If Localstack ever changes to adopt the correct S3 behavior, I guess we'd still always use your new logic here so this test might not actually help us in that case. Does that make sense? I could be missing the point.

Reply:
Hmm, interesting thought. I'm inclined to say, like you did, that we don't want the unit test for S3ProviderClient.sync to know which code path is hit necessarily. In some cases we do assert something about the code path for a unit test because creating the system state necessary for a unit test is impractical, so mocking and verifying that the mock was used is preferable. But in this case we're actually staging a fixture for a 0 byte file and proving that `sync` handles it properly. 

I don't know if that answers the question

Reply:
Yeah, I think what you have here is right and the alternative is ...undesirable. It just feels a little wonky that we could revert your code changes and this test would still pass because Localstack's behavior doesn't match the real deal.

This test 100% does what it says it's going to though: `S3ProviderClient.sync syncs a 0 byte file` and there's an integration test to double-check in a deployed environment.

I think it's good. I can't think of a non-wonky way to validate that your workaround specifically works and I'm not sure we need to validate that here.